src/modules/household/hooks/useChangeHoh.tsx

```diff
@@ -0,0 +1,108 @@
+import { useCallback, useMemo, useState } from 'react';
+
+import ValidationDialog from '@/modules/errors/components/ValidationDialog';
+import {
+  ErrorState,
+  emptyErrorState,
+  partitionValidations,
+} from '@/modules/errors/util';
+import {
+  HouseholdClientFieldsFragment,
+  RelationshipToHoH,
+  useUpdateRelationshipToHoHMutation,
+} from '@/types/gqlTypes';
+
+interface Args {
+  refetchHousehold: VoidFunction;
+}
+
+// hook with logic for changing the HoH, and confirmation dialog
+export function useChangeHoh({ refetchHousehold }: Args) {
+  const [proposedHoh, setProposedHoh] =
+    useState<HouseholdClientFieldsFragment | null>(null);
+  const [warningsConfirmed, setWarningsConfirmed] = useState(false);
+  const [errorState, setErrors] = useState<ErrorState>(emptyErrorState);
+  // client to highlight for relationship input
+  const [highlight, setHighlight] = useState<string[]>([]);
+
+  const resetState = useCallback(() => {
+    setProposedHoh(null);
+    setErrors(emptyErrorState);
+    setWarningsConfirmed(false);
+  }, []);
+
+  const [setHoHMutate, { loading: hohChangeLoading }] =
```

Comment:
This is really minor but, in case it's easy to fix: I noticed that visually when changing the HoH, while the hhm table is reloading, there are 2 HoH checkmarks. I wonder if this might give the user the wrong impression that we add HoH status to another member before removing it from the original HoH. Of course in reality this happens in a transaction so, nothing to worry about.

Reply:
Thanks, thats a good point. There definitely would be ways to do it on the frontend (like: set the new hoh in state and pass it around; or write the updated relationships to the apollo cache) but the cleanest way, I think, would be to adjust the `UpdateRelationshipToHoH` mutation to return the full household, so relationship can be updated in the cache as soon as the mutation completes. I'm going to hold off on that to keep this to frontend-only changes.